### PR TITLE
Machine API updates for 4.5 RNs

### PR DESCRIPTION
**Pull request by codyhoag**
Adds new features to the 4.5 Release Notes outlined in https://github.com/openshift/openshift-docs/issues/19847#issuecomment-651163656.

Preview: http://file.rdu.redhat.com/~choag/063020/rn-4.5-machine-api/release_notes/ocp-4-5-release-notes.html#ocp-4-5-machine-api

### CODE REVIEW

**Comment by codyhoag** [comment]
@enxebre can you verify this info for the release notes? Thanks!

File: release_notes/ocp-4-5-release-notes.adoc [hunk 1]
```diff
@@ -101,6 +101,44 @@ type of private DNS zone.
...
+----
+providerSpec:
+  spotMarketOptions: 
+----
+
+[id="ocp-4-5-autoscaling-the-minimum-number-of-machines"]
+==== Autoscaling the minimum number of machines to 0
+
+For clusters running on AWS, Azure, and GCP, you can now set the minimum number
+of machines that must remain in the specified zone to `0`. This setting applies
```
Comment (enxebre) [inline release_notes/ocp-4-5-release-notes.adoc:125]: I think this needs clarification:
You can now set the minimum number of replicas for a `machineAutoscaler` to 0.
This enable the autoscaler to be even more cost effective by being able to scale from zero machines to anything and getting back to zero based on the resource your workloads require.

**Comment by enxebre** [comment]
other than https://github.com/openshift/openshift-docs/pull/23389/files#r449063258 lgtm
PTAL @JoelSpeed

File: release_notes/ocp-4-5-release-notes.adoc [hunk 2]
```diff
@@ -101,6 +101,44 @@ type of private DNS zone.
...
+
+AWS MachineSets now support spot instances. This lets you create a MachineSet
+that deploys machines as spot instances so you can save costs compared to
+on-demand instance prices. You can configure spot instances by adding the
+following line under the `providerSpec` field in the MachineSet YAML file:
+
+[source,yaml]
+----
+providerSpec:
+  spotMarketOptions:
```
Comment (JoelSpeed) [inline release_notes/ocp-4-5-release-notes.adoc:118]: I don't think leaving the line trailing empty is valid yaml is it? I've always done it as
```suggestion
  spotMarketOptions: {}
```

**Review by JoelSpeed — COMMENTED** [review]
Looks good, just one minor thing on the spot instance example, I don't think it will work as it is currently

File: release_notes/ocp-4-5-release-notes.adoc [hunk 3]
```diff
@@ -101,6 +101,44 @@ type of private DNS zone.
...
+----
+providerSpec:
+  spotMarketOptions: {}
+----
+
+[id="ocp-4-5-autoscaling-the-minimum-number-of-machines"]
+==== Autoscaling the minimum number of machines to 0
+
+You can now set the minimum number of replicas for a MachineAutoscaler to `0`.
+This allows the autoscaler to be more cost effective by scaling between zero
```
Comment (bobfuru) [inline release_notes/ocp-4-5-release-notes.adoc:125]: s/cost effective/cost-effective/

**Review by bobfuru — COMMENTED** [review]
One tiny nit and LGTM!

**Pull request merged**